### PR TITLE
refactor chm_diags and het_diags

### PR DESCRIPTION
**Pull request by tangshuaiqi**
1. remove unused code related to unused species
2. rename some variables to more meaningful
3. change the way of calculating tropospheric and stratospheric ozone
4. some other regular refactoring.

as the two subroutines are for diagnostics with many outfld calls. I have run long-term test to make sure outfld test also passed:

comp_testall_mo_chm_diags_mam4_org_v2_baselines: 9 tests
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P1x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

### CODE REVIEW

File: components/eam/src/chemistry/mozart/mo_chm_diags.F90 [hunk 1]
```diff
@@ -388,17 +390,18 @@ subroutine chm_diags( lchnk, ncol, vmr, mmr, rxt_rates, invariants, depvel, depf
     !--------------------------------------------------------------------
     !	... local variables
     !--------------------------------------------------------------------
-    integer     :: i,j,k, m, n
-    integer :: plat
-    real(r8)    :: wrk(ncol,pver)
-    real(r8)    :: un2(ncol)
+    integer     :: icol,kk, mm, nn
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/mozart/mo_chm_diags.F90:263]: I think we can get rid of all `#if defs` for `MODAL_AERO`

Comment (singhbalwinder) [inline components/eam/src/chemistry/mozart/mo_chm_diags.F90:263]: `rxn_rates` and  `invariants` are not used here

File: components/eam/src/chemistry/mozart/mo_chm_diags.F90 [hunk 2]
```diff
@@ -388,17 +390,18 @@ subroutine chm_diags( lchnk, ncol, vmr, mmr, rxt_rates, invariants, depvel, depf
...
+    integer     :: plat
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/mozart/mo_chm_diags.F90]: plat is never used

File: components/eam/src/chemistry/mozart/mo_chm_diags.F90 [hunk 3]
```diff
@@ -753,8 +607,8 @@ subroutine het_diags( het_rates, mmr, pdel, lchnk, ncol )
     real(r8), intent(in)  :: pdel(ncol,pver)
 
     real(r8), dimension(ncol) :: noy_wk, sox_wk, nhx_wk, wrk_wd
-    integer :: m, k, j
-    integer :: plat
+    integer  :: mm, kk
+    integer  :: plat
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/mozart/mo_chm_diags.F90]: plat is never used here

**Review by singhbalwinder — CHANGES_REQUESTED** [review]
Looks good! These routines are computing just the diagnostics, so nothing much to do here. I liked the way you have changed variable names and named them appropriately.  Left some very minor comments.

**Comment by tangshuaiqi** [comment]
@singhbalwinder all comments have been addressed. I also removed unused variables and indices in the module variables and in chm_diags_init.

**Comment by tangshuaiqi** [comment]
All tests passed:
comp_testall_mo_chm_diags_mam4_org_v2_baselines: 9 tests
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P1x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Review by singhbalwinder — APPROVED** [review]
This is ready to go! Thanks @tangshuaiqi !

**Pull request merged**